giraffe/src/utils/identityMerge.ts

```diff
@@ -128,7 +128,11 @@ export const setByPath = (target: any, path: Path, value: any): void => {
     currentTarget = nextTarget
   }
 
-  currentTarget[path[i + 1]] = value
+  // never write into the input config, Object.freeze might have been applied
```

Comment:
The code is fine, but this comment can be confusing. "never" is not really the case here. I think the code means, if the value hasn't changed, do not write into the config. Because when the value has changed (being not equal), there will be writing into the config.

Reply:
When the value has changed (being not equal), there will be writing into the config, but not the input one. Rather the 'prevConfig' as referenced in the code if I remember properly.

I have changed the comment to be easier to understand anyway.